lib/rex/proto/http/client_request.rb

```diff
@@ -136,12 +136,16 @@ def to_s
 
       opts['vars_post'].each_pair do |var,val|
         var = var.to_s
-        val = val.to_s
-
-        pstr << '&' if pstr.length > 0
-        pstr << (opts['encode_params'] ? set_encode_uri(var) : var)
-        pstr << '='
-        pstr << (opts['encode_params'] ? set_encode_uri(val) : val)
+        unless val.is_a?(Array)
+          val = val.split
```

Comment:
This is a breaking change since spaces that would get encoded before would now trigger sending multiple values to a server. A better way might be
```suggestion
          val = [val]
```